kubeflow/common/iap-ingress/base/config-map.yaml

```diff
@@ -181,6 +179,91 @@ data:
+
+    PROJECT=$(curl -s -H "Metadata-Flavor: Google" http://metadata.google.internal/computeMetadata/v1/project/project-id)
+    if [ -z ${PROJECT} ]; then
+      echo Error unable to fetch PROJECT from compute metadata
+      exit 1
+    fi
+
+    PROJECT_NUM=$(curl -s -H "Metadata-Flavor: Google" http://metadata.google.internal/computeMetadata/v1/project/numeric-project-id)
+    if [ -z ${PROJECT_NUM} ]; then
+      echo Error unable to fetch PROJECT_NUM from compute metadata
+      exit 1
+    fi
+
+    # Activate the service account
+    if [ ! -z "${GOOGLE_APPLICATION_CREDENTIALS}" ]; then
+      # As of 0.7.0 we should be using workload identity and never setting GOOGLE_APPLICATION_CREDENTIALS.
+      # But we kept this for backwards compatibility but can remove later.
+      gcloud auth activate-service-account --key-file=${GOOGLE_APPLICATION_CREDENTIALS}
+    fi
+
+    # Print out the config for debugging
+    gcloud config list
+    gcloud auth list
+
+    set_endpoint () {
+      NODE_PORT=$(kubectl --namespace=${NAMESPACE} get svc ${SERVICE} -o jsonpath='{.spec.ports[?(@.name=="http2")].nodePort}')
+      echo "[DEBUG] node port is ${NODE_PORT}"
+
+      BACKEND_NAME=""
+      while [[ -z ${BACKEND_NAME} ]]; do
+        BACKENDS=$(kubectl --namespace=${NAMESPACE} get ingress ${INGRESS_NAME} -o jsonpath='{.metadata.annotations.ingress\.kubernetes\.io/backends}')
+        echo "[DEBUG] fetching backends info with ${INGRESS_NAME}: ${BACKENDS}"
+        BACKEND_NAME=$(echo $BACKENDS | grep -o "k8s-be-${NODE_PORT}--[0-9a-z]\+")
+        echo "[DEBUG] backend name is ${BACKEND_NAME}"
+        sleep 2
+      done
+
+      BACKEND_ID=""
+      while [[ -z ${BACKEND_ID} ]]; do
+        BACKEND_ID=$(gcloud compute --project=${PROJECT} backend-services list --filter=name~${BACKEND_NAME} --format='value(id)')
+        echo "[DEBUG] Waiting for backend id PROJECT=${PROJECT} NAMESPACE=${NAMESPACE} SERVICE=${SERVICE} filter=name~${BACKEND_NAME}"
+        sleep 2
+      done
+      echo BACKEND_ID=${BACKEND_ID}
+
+      JWT_AUDIENCE="/projects/${PROJECT_NUM}/global/backendServices/${BACKEND_ID}"
+      INGRESS_TARGET_IP=$(kubectl get ingress --all-namespaces | grep -E -o "(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)")
+      echo "[DEBUG] ENDPOINT_NAME = ${ENDPOINT_NAME}"
+      echo "[DEBUG] INGRESS_TARGET_IP = ${INGRESS_TARGET_IP}"
+      echo "[DEBUG] JWT_AUDIENCE = ${JWT_AUDIENCE}"
+      sed "s|JWT_AUDIENCE|${JWT_AUDIENCE}|;s|ENDPOINT_NAME|${ENDPOINT_NAME}|;s|INGRESS_TARGET_IP|${INGRESS_TARGET_IP}|" /var/envoy-config/swagger_template.yaml > openapi.yaml
+      
+      gcloud endpoints services deploy openapi.yaml
+
+      gcloud services enable servicemanagement.googleapis.com
+      gcloud services enable servicecontrol.googleapis.com
+      gcloud services enable endpoints.googleapis.com
```

Comment:
Can these be run beforehand? We have this section in the deployment guide: https://www.kubeflow.org/docs/distributions/gke/deploy/project-setup/#set-up-project-and-api-scopes

Reply:
This is a good point. We can add `servicecontrol.googleapis.com` and `endpoints.googleapis.com` to the page you mentioned.

Reply:
https://github.com/kubeflow/website/pull/3316

Reply:
Awesome! Thank you for the changes!